<launch>
  <!-- Switch from centering command to graph planner -->
  <node pkg="graph-planner" type="junction_twistmsg_switch.py" name="cmd_vel_switch" output="screen">
    <!-- Inputs (Subscribed Topic Names) -->
    <!-- First and second command velocities -->
    <remap from="cmd_vel_1" to="/X1/cmd_vel_centering"/>
    <remap from="cmd_vel_2" to="/X1/cmd_vel_junction"/>
    
    <!-- Whether to stay on cmd_vel_1 (False) or to switch to cmd_vel_2 (True) -->
    <remap from="switch" to="/X1/at_a_node"/>

    <!-- Outputs (Published Topic Names) -->
    <!-- The topic to publish cmd_vel_1 or cmd_vel_2 to -->
    <remap from="cmd_vel" to="/X1/cmd_vel"/>

    <!-- Parameters -->
    <!-- Update rate (Hz) -->
    <param name="rate" value="3.0"/>
  </node>
</launch>